### PR TITLE
Do not allow adding device when device name is too long

### PR DESCRIPTION
**Pull request by mchf**
bnc#991486

### CODE REVIEW

**Comment by teclator** [comment]
It would be nice to have a test, in that case LGTM

**Comment by mchf** [comment]
Note ```ValidNicName``` is already covered by test

**Comment by teclator** [comment]
I mean, `validate_hw` but is ok, just being picky

**Comment by mchf** [comment]
yes, I got it ;-) That was note mainly for me and my friend Alzheimer

File: test/hardware_test.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  it "passes for valid names only" do
+    allow(subject).to receive(:devname_from_hw_dialog).and_return valid_name
+
+    expect(subject.validate_hw(nil, nil)).to be true
+  end
+
+  # bnc#991486
+  it "fails for long names" do
+    allow(subject).to receive(:devname_from_hw_dialog).and_return long_name
```
Comment (teclator) [inline test/hardware_test.rb]: you also could check that in this case it should set the focus

File: test/hardware_test.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  let(:valid_name) { "eth0" }
+  let(:long_name) { "verylongcustomnicname123" }
+
+  before(:each) do
+    allow(Yast::UI).to receive(:SetFocus)
+    expect(subject).to receive(:UsedNicName).and_return false
+  end
+
+  it "passes for valid names only" do
```
Comment (teclator) [inline test/hardware_test.rb]: it fails for valid names and also for used nic names

**Comment by teclator** [comment]
LGTM

**Pull request merged**